src/vector/feature.rs

```diff
@@ -55,6 +72,54 @@ impl<'a> Feature<'a> {
         }
         return &self.geometry;
     }
+
+    pub fn create(&self, lyr: &Layer) -> Result<()> {
+        let rv = unsafe { ogr::OGR_L_CreateFeature(lyr.gdal_object_ptr(), self.c_feature) };
+        if rv != ogr_enums::OGRErr::OGRERR_NONE {
+            return Err(ErrorKind::OgrError(rv, "OGR_L_CreateFeature").into());
+        }
+        Ok(())
+    }
+
+    pub fn set_field_string(&self, field_name: &str, value: &str) -> Result<()> {
+        let c_str_field_name = CString::new(field_name)?;
+        let c_str_value = CString::new(value).unwrap();
+        let idx = unsafe { ogr::OGR_F_GetFieldIndex(self.c_feature, c_str_field_name.as_ptr())};
+        unsafe { ogr::OGR_F_SetFieldString(self.c_feature, idx, c_str_value.as_ptr()) };
+        Ok(())
+    }
+
+    pub fn set_field_double(&self, field_name: &str, value: f64) -> Result<()> {
+        let c_str_field_name = CString::new(field_name)?;
+        let idx = unsafe { ogr::OGR_F_GetFieldIndex(self.c_feature, c_str_field_name.as_ptr())};
+        unsafe { ogr::OGR_F_SetFieldDouble(self.c_feature, idx, value as c_double) };
+        Ok(())
+    }
+
+    pub fn set_field_integer(&self, field_name: &str, value: i32) -> Result<()> {
+        let c_str_field_name = CString::new(field_name)?;
+        let idx = unsafe { ogr::OGR_F_GetFieldIndex(self.c_feature, c_str_field_name.as_ptr())};
+        unsafe { ogr::OGR_F_SetFieldInteger(self.c_feature, idx, value as c_int) };
+        Ok(())
+    }
+
+    pub fn set_field(&self, field_name: &str, type_value: OGRFieldType, value: FieldValue) -> Result<()> {
+        match type_value {
+            OGRFieldType::OFTReal => self.set_field_double(field_name, value.to_real()),
```

Comment:
Shouldn't the `Result` from `set_field_double` (and others) just be passed to the caller of `set_field` like this?

```rust
pub fn set_field(&self, field_name: &str,  value: FieldValue) -> Result<()> {
      match value {
         FieldValue::RealValue(value) => self.set_field_double(field_name, value),
         FieldValue::StringValue(value) => self.set_field_string(field_name, value.as_str()),
         FieldValue::IntegerValue(value) => self.set_field_integer(field_name, value)
     }
 }
```
This way, with an invalid field name we get the `ErrKind::InvalidFieldName` error from the `set_field_double/string/..` function, and we can't enter an invalid FieldValue due to the type checking.